TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/kafka/KafkaProtobufConsumer.java

```diff
@@ -20,25 +24,108 @@
 import java.time.Duration;
 import java.util.Collections;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
+import java.util.Optional;
+import java.util.PriorityQueue;
 import java.util.Properties;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import java.util.stream.StreamSupport;
 
+/**
+ * Adapt a Kafka stream into a TrafficCaptureSource.
+ *
+ * Notice that there's a critical gap between how Kafka accepts commits and how the
+ * BlockingTrafficSource throttles calls to Kafka.  The BlockingTrafficSource may
+ * block calls to readNextTrafficStreamChunk() until some time window elapses.  This
+ * could be a very large window in cases where there were long gaps between recorded
+ * requests from the capturing proxy.  For example, if a TrafficStream is read and it
```

Comment:
It feels like I'm missing a chunk of text that got deleted at the end of this line

Reply:
As I wrote it, I figured out a fix. I plan on patching this shortly and amending the comments. 
See https://opensearch.atlassian.net/browse/MIGRATIONS-1428